src/Addon/Plugin/PluginCriteria.php

```diff
@@ -198,8 +213,8 @@ public function getCacheKey()
             $this->options,
             'cache.key',
             $this->getCachePrefix() . '.'
-            . md5(json_encode($this->collection)) . '.'
-            . md5(json_encode($this->options))
+            . md5(json_encode($this->options)) . '.'
+            . md5(json_encode($this->getCallbackStaticVariables()))
```

Comment:
`$this->getCallbackStaticVariables()` gets the variables that are passed threw a `use` statement.
For example the `$menu` variable here: https://github.com/anomalylabs/navigation-module/blob/2.4/src/NavigationModulePlugin.php#L33
Or the `$method` variable here: https://github.com/anomalylabs/twitter-extension/blob/1.0/src/TwitterExtensionPlugin.php#L31

This is important, because these are the original twig plugin paramters.
I came across several cases where I got the same cache key for different requests because this was missing.